src/streamlink/plugins/tvp.py

```diff
@@ -169,8 +172,58 @@ def _get_vod(self, vod_id):
         if data.get("HLS"):
             return HLSStream.parse_variant_playlist(self.session, data["HLS"][0]["src"])
 
+    def _get_tvp_info_vod(self):
+        data = self.session.http.get(
+            self.url,
+            schema=validate.Schema(
+                validate.parse_html(),
+                validate.xml_xpath_string(".//script[contains(text(), 'window.__videoData')]"),
+                validate.none_or_all(
+                    str,
+                    re.compile(r"window\.__videoData\s*=\s*(?P<json>{.*?});", re.DOTALL),
+                    validate.get("json"),
```

Comment:
The validation of `re.Pattern` can return `None`, so the `get()` call afterwards will fail with a bad validation error message from the raised `AttributeError`.
https://github.com/streamlink/streamlink/blob/5.1.2/src/streamlink/plugin/api/validate/_validate.py#L135

Use `validate.regex(re.Pattern)` if you want to ensure that a compiled pattern matches. If it doesn't, then a proper validation error message will be raised.
https://github.com/streamlink/streamlink/blob/5.1.2/src/streamlink/plugin/api/validate/_validate.py#L216

The regex itself could probably be improved by adding `\s*` in between the end of the JSON payload and the semicolon character.

Reply:
Please also update the other schemas. I made a mistake here when I rewrote the plugin a couple of months ago.

- https://github.com/streamlink/streamlink/blame/5.1.2/src/streamlink/plugins/tvp.py#L45
- https://github.com/streamlink/streamlink/blame/5.1.2/src/streamlink/plugins/tvp.py#L94